authentication/auth.go

```diff
@@ -142,19 +141,19 @@ type QuestionnaireSchema struct {
 	FormType string `json:"form_type"`
 }
 
-// Generates a random string of a defined length
-func randomStringGen() (rs string) {
-	size := 6 //change the length of the generated random string
-	rb := make([]byte, size)
-	_, err := rand.Read(rb)
+// Generates something that looks like a UUID4 but doesn't entirely confirm to spec
+func pseudoUUID() (uuid string) {
 
-	if err != nil {
-		log.Println(err)
-	}
+    b := make([]byte, 16)
+    _, err := rand.Read(b)
+    if err != nil {
+        fmt.Println("Error: ", err)
+        return
+    }
 
-	randomString := base64.URLEncoding.EncodeToString(rb)
+    uuid = fmt.Sprintf("%X-%X-%X-%X-%X", b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
```

Comment:
Why don't we use https://godoc.org/github.com/satori/go.uuid

Reply:
There were 3/4 uuid libraries, none of them seemed particularly official looking.